File: 202201/ValidateNumber.java
class ValidateNumber{
    public static final boolean validaNumericString(String numStr){
        int expCount = 1;
        int decCount = 1;
        int sign_count = 1;
        for(int i=0;i<numStr.length();i++){
            char c = numStr.charAt(i);
            if(Character.isDigit(c))
                continue;
            else if( c == '+' || c == '-'){
                if( i == 0 & sign_count > 0)
                    sign_count -= 1;
                else
                    return false;
            }
            else if(c == '.' & i != 0){
                if(decCount > 0){
                    decCount -= 1;
                }
                else
                    return false;
            }
            else if( c == 'e' & i != 0){
                if(expCount > 0)
                    expCount -= 1;
                else
                    return false;
            }
        }
        return true;
    }

    public static void main(String[] args) {
        String[] tests = {
            "10.1",
            "-10.1",
            "1e5",
            "1.025e10",
            "1e10.25"
        };

        for(String t: tests){
            System.out.println(String.format("%s %b", t, ValidateNumber.validaNumericString(t)));
        }
    }
}

File: 202201/ValidateNumber.py
class ValidateNumber:
    def validate(self, num_str: str) -> bool:
        dec_count = 1
        exp_count = 1
        for i,c in enumerate(num_str):
            if c.isdigit():
                continue
            elif c in ['-', '+']:
                if i != 0:
                    return False
            elif c == '.':
                if dec_count and i != 0 and exp_count > 0:
                    dec_count -= 1
                else:
                    return False
            elif c.lower() == 'e':
                if exp_count and i != 0:
                    exp_count -= 1
                else:
                    return False
            else:
                return False
        return True



tests = [
    '10',
    '-10',
    '-10.01',
    '1e10',
    '1.02e25',
    '1e0.25',
    '1.2.5urn'
]


for t in tests:
    print("{} : {}".format(t, ValidateNumber().validate(t)))
            
                
            